app/controllers/Application.scala

```diff
@@ -84,6 +85,51 @@ class Application @Inject()(files: FileService, collections: CollectionService,
     }
   }
 
+
+  /**
+   * Returns the sitemap.xml for the datasets to be scraped for their jsonld scripts
+   * suggested to start like w/swagger route, but if I don't cache it, then I should change this
+   *  otherwise it will need a filler file there; which I should provide as a cache
+   */
+  def sitemap = Action { implicit request =>
+    Play.resource("/public/sitemap.xml") match { //in case we cache it here someday
+      case Some(resource) => {
+        val https = Utils.https(request)
+        val clowderurl = new URL(Utils.baseUrl(request))
+        val host = if (clowderurl.getPort == -1) {
+          clowderurl.getHost
+        } else {
+          clowderurl.getHost + ":" + clowderurl.getPort
+        }
+        var resultStr=""
+        val top= """<?xml version="1.0" encoding="UTF-8"?>
+            <urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9"> """
+        resultStr = resultStr.concat(top)
+        //though had called the route2get but couldn't change datastruct
+        val d = scala.io.Source.fromURL(clowderurl + "/api/datasets")
```

Comment:
why not just call the code directly? `datasets.list()`

Reply:
That got lost, will get it in there, tnx

Reply:
doing a getDatasets for User.anaoymous for now
can look at possible caching, variable list lengths in next iteration